### PR TITLE
Сustomizing Electric Bolt Arcs

### PR DESCRIPTION
**Pull request by Fryone**
By default, Electric Bolt has 8 Arcs. Now it can be customized per weapon with `IsElectricBolt=yes`
Here is example of Tesla Coil using 1 arc for primary and 32 for secondary weapon.

https://github.com/Phobos-developers/Phobos/assets/30149531/7a88a52c-e26d-44e3-b812-5e698ec0e639

Created with big help from Kerbiter

### CODE REVIEW

File: src/Ext/WeaponType/Hook.EBolt.cpp [hunk 1]
```diff
@@ -27,6 +29,19 @@ DEFINE_HOOK(0x4C2951, EBolt_DTOR, 0x5)
...
 }
 
+DEFINE_HOOK(0x4C20BC, EBolt_DrawArcs, 0xB)
+{
+	enum { DoLoop = 0x4C20C7, Break = 0x4C2400 };
+
+	GET_STACK(EBolt*, pBolt, 0x40);
+	BoltTemp::pType = BoltTemp::boltWeaponTypeExt.get_or_default(pBolt);
+
+	GET_STACK(int, plotIndex, STACK_OFFSET(0x408, -0x3E0))
```
Comment (MortonPL) [inline src/Ext/WeaponType/Hook.EBolt.cpp:39]:
```suggestion
	GET_STACK(EBolt*, pBolt, 0x40);
	GET_STACK(int, plotIndex, STACK_OFFSET(0x408, -0x3E0));
	BoltTemp::pType = BoltTemp::boltWeaponTypeExt.get_or_default(pBolt);
```

In most cases we group GET macros together.

**Review by MortonPL — CHANGES_REQUESTED** [review]
Please add an entry in credits, "What's New" and "Fixes & Improvements".

**Comment by mevitar** [comment]
If default value is 8, then i think i the documentation should have Bolt.Arcs=8 instead Bolt.Arcs=32 as an example. I'm pretty sure all other entries in the doc follow that formula.

BTW, is there a maximum value? I set Bolt.Arcs=64 and even went into several hundreds, expecting to get something crazy, but couldn't really see any difference. The only noticeable differences are between 0, 1 and default, with anything in between hardly noticeable, and anything beyond default not noticeable at all.

File: docs/Fixed-or-Improved-Logics.md [hunk 2]
```diff
@@ -789,6 +789,17 @@ Bolt.Disable2=false    ; boolean
...
 
+### Customizable ElectricBolt Arcs
+
+- By default, Electric Bolt has 8 Arcs. Now it can be customized per weapon with `IsElectricBolt=yes`. Zero value draws straight line.
+
+In `rulesmd.ini`:
+```ini
+[SOMEWEAPONTYPE]       ; WeaponType
+IsElectricBolt=true    ; an ElectricBolt Weapon, vanilla tag
+Bolt.Arcs=32
```
Comment (MortonPL) [inline docs/Fixed-or-Improved-Logics.md:800]:
```suggestion
IsElectricBolt=true    ; boolean, vanilla tag
Bolt.Arcs=8            ; integer, number of arcs in a bolt
```

**Review by MortonPL — APPROVED** [review]
Looks good, just needs one more small change in documentation.

**Review by Metadorius — APPROVED** [review]

**Review by Fryone — COMMENTED** [review]
I approve

**Pull request merged**